addons/website_slides/static/tests/tours/slides_tour_tools.js

```diff
@@ -322,12 +329,78 @@ var addNewCourseTag = function (courseTagName, backend) {
 }];
 };
 
+/**
+ * Adds a new quiz to the given section name.
+ */
+const addQuizToSection = (sectionName, quizName, addQuestionsOptions = [], backend) => {
+    const prefix = backend ? ':iframe ' : '';
+    let quizSteps = [
+{
+    content: 'eLearning: add content to section',
+    trigger: fetchContentSelectorFromSection(prefix, sectionName),
+    run: "click",
+}, {
+    content: 'eLearning: click on quiz',
+    trigger: prefix + 'a[data-slide-category=quiz]',
+    run: "click",
+}, {
+    content: 'eLearning: fill quiz title',
+    trigger: prefix + 'input[name=name]',
+    run: `edit ${quizName}`,
+}, {
+    content: 'eLearning: fill quiz completion time',
+    trigger: prefix + 'input[name=duration]',
+    run: "edit 10",
+}, {
+    content: 'eLearning: create and publish slide',
+    trigger: prefix + 'footer.modal-footer button:contains("Publish")',
+    run: "click",
+}];
+if (addQuestionsOptions.length) {
+    const addQuestionSteps = addQuestionsOptions.flatMap((questionOption, i, questions) => {
+        const steps = [
+            {
+                content: 'eLearning: add a question',
+                trigger: ':iframe input[name="question-name"]',
+                run: `edit ${questionOption.question}`,
+            },
+            ...questionOption.options.flatMap((option, index) => [
+                {
+                    content: `eLearning: enter option ${index + 1}`,
+                    trigger: `:iframe div.list-group .o_wslides_js_quiz_answer:nth-of-type(${index + 1}) input`,
+                    run: `edit ${option.title}`,
+                },
+                ...(option.isCorrect ? [{
+                    content: 'eLearning: select the correct option',
+                    trigger: `:iframe div.list-group .o_wslides_js_quiz_answer:nth-of-type(${index + 1}) i`,
+                    run: "click",
+                }] : [])
+            ]),
+            // on adding the last question, we only click the SAVE button
+            i === questions.length - 1 ? {
```

Comment:
Hello @flch-odoo :wave: ,
I am almost done with the changes, the only point where I am stuck is to check for the absence of an element. :frowning_face: 

My tour step----
```js
content: "eLearning: should not be able to input the title of a new question.",
trigger: ':iframe .o_wslides_js_lesson_quiz',
run: () => {
    expectHiddenElement(
        '.o_wsildes_quiz_question_input',
        'We should not be able to input a new question'
    );
},
```

The function ---
```js
export function expectHiddenElement(selector, msg) {
    if (queryAll(`.o_iframe:iframe ${selector}`).length) {
        throw new TourError(msg);
    }
}
```

Don't know why but everytime I run the tour from the backend, I get an error.

The error is ---
`Tour course_publisher_standard failed at step eLearning: should not be able to input the title of a new question. (trigger: :iframe .o_wslides_js_lesson_quiz). Element has been found. The error seems to be with step.run`

Reply:
Indeed, to not create race conditions we look for elements with a `d-none` class, (not assert that there is none without) as we did in the mentioned function. 
However, it does not work here because the form gets destroyed, sorry for misleading you on this one.

It seems however that we could expect to find the `o_wslides_js_quiz_add_question` button with `d-none` when the form is visible and removed when the form is destroyed. Wouldn't this work as trigger?

Reply:
@flch-odoo 

I tried to find the trigger `.o_wslides_js_quiz_add_question` after we have created 2 questions and when `+ Add Question` button is visible.

But still it was not found in the DOM.

My plan was to do this---

```js
content: "eLearning: should not be able to input the title of a new question.",
trigger: ':iframe .o_wslides_js_quiz_add_question:not(.d-none)',
isCheck: true,
```

Reply:
I would have thought `${prefix}a.o_wslides_js_quiz_add_question:not(.d-none)` would work...

It it doesn't, could you please push your most recent version?

Reply:
Here you go :+1:

Reply:
Found it :)
After clicking on "Save & New", you have to wait for the widget to be destroyed and reinitialized.
What happens in your test is that you're able to edit the question title *of the initial form title before it is destroyed*, and so you end up with an empty question field on your new widget, and you are unable to save that question.
As you are able to click the button, we only see that we can't get to pass the following step, but that one is fine.

Something like this does the trick.
```js
        ... (i === questions.length - 1 ? [{
            content: 'eLearning: save the question',
            trigger: ':iframe a.o_wslides_js_quiz_validate_question',
            run: "click",
        }]: [{
            content: 'eLearning: click SAVE & NEW button',
            trigger: ':iframe .o_wslides_js_quiz_validate_question_and_new',
            run: "click",
        }, {
            trigger: `${prefix}.o_wslides_quiz_question_sequence:contains(${i + 2})`
        }]),
```

Let me know when the PR is ready for the next check :)

Reply:
Thanks a lot for the help @flch-odoo :pray: 
I have updated the code :+1:

Reply:
I looked at the screenshot of the failure which showed something weird, then I ran the test tour with `watch=True` to see it happen and have the full page visible.